planner/core/task.go

```diff
@@ -819,22 +820,41 @@ func (p *PhysicalLimit) attach2Task(tasks ...task) task {
 	t := tasks[0].copy()
 	sunk := false
 	if cop, ok := t.(*copTask); ok {
-		// For double read which requires order being kept, the limit cannot be pushed down to the table side,
-		// because handles would be reordered before being sent to table scan.
-		if (!cop.keepOrder || !cop.indexPlanFinished || cop.indexPlan == nil) && len(cop.rootTaskConds) == 0 {
-			// When limit is pushed down, we should remove its offset.
-			newCount := p.Offset + p.Count
-			childProfile := cop.plan().statsInfo()
-			// Strictly speaking, for the row count of stats, we should multiply newCount with "regionNum",
-			// but "regionNum" is unknown since the copTask can be a double read, so we ignore it now.
-			stats := deriveLimitStats(childProfile, float64(newCount))
-			pushedDownLimit := PhysicalLimit{Count: newCount}.Init(p.ctx, stats, p.blockOffset)
-			cop = attachPlan2Task(pushedDownLimit, cop).(*copTask)
-			// Don't use clone() so that Limit and its children share the same schema. Otherwise the virtual generated column may not be resolved right.
-			pushedDownLimit.SetSchema(pushedDownLimit.children[0].Schema())
+		if len(cop.idxMergePartPlans) == 0 {
+			// For double read which requires order being kept, the limit cannot be pushed down to the table side,
+			// because handles would be reordered before being sent to table scan.
+			if !cop.keepOrder || !cop.indexPlanFinished || cop.indexPlan == nil || len(cop.rootTaskConds) == 0 {
+				// When limit is pushed down, we should remove its offset.
+				newCount := p.Offset + p.Count
+				childProfile := cop.plan().statsInfo()
+				// Strictly speaking, for the row count of stats, we should multiply newCount with "regionNum",
+				// but "regionNum" is unknown since the copTask can be a double read, so we ignore it now.
+				stats := deriveLimitStats(childProfile, float64(newCount))
+				pushedDownLimit := PhysicalLimit{Count: newCount}.Init(p.ctx, stats, p.blockOffset)
+				cop = attachPlan2Task(pushedDownLimit, cop).(*copTask)
+				// Don't use clone() so that Limit and its children share the same schema. Otherwise the virtual generated column may not be resolved right.
+				pushedDownLimit.SetSchema(pushedDownLimit.children[0].Schema())
+			}
+			t = cop.convertToRootTask(p.ctx)
+			sunk = p.sinkIntoIndexLookUp(t)
+		} else if !cop.idxMergeIsIntersection {
```

Comment:
Should we add a switch for this feature?

Reply:
I'm thinking about whether to add one